addons/website/migrations/12.0.1.0/pre-migration.py

```diff
@@ -1,11 +1,203 @@
 # Copyright 2019 Eficent <http://www.eficent.com>
 # Copyright 2019 Tecnativa - Pedro M. Baeza
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from base64 import b64encode
+from itertools import chain, product
+from os import listdir
+from pathlib import Path
+
+from psycopg2.extensions import AsIs
+
 from openupgradelib import openupgrade
+from openupgradelib.openupgrade_120 import convert_string_bootstrap_3to4
 
 _field_renames = [
     ('website.page', 'website_page', 'website_published', 'is_published'),
 ]
+_xml_ids_renames = [
+    ('website.menu_homepage', 'website.menu_home'),
+]
+
+
+def _apply_v11_dummy_image_replacement(replacements, original_html):
+    """Perform image URL replacements."""
+    new_html = original_html
+    for xmlid, dbid in replacements.items():
+        old_url = "/web/image/%s" % xmlid
+        new_url = "/web/image/%d" % dbid
+        new_html = new_html.replace(old_url, new_url)
+    return new_html
+
+
+def _preserve_v11_dummy_images(env):
+    """Preserve dummy images, which could still be used in some websites."""
+    replacements = {}
+    images_dir = Path(__file__).parent / "changed_dummy_images"
+    for image_path in map(images_dir.joinpath, listdir(str(images_dir))):
+        xmlid = image_path.stem
+        try:
+            old = env.ref(xmlid)
+        except ValueError:
+            continue  # Attachment was deleted; nothing to do
+        with image_path.open("rb") as image_stream:
+            new = old.copy({
+                "url": False,
+                "datas": b64encode(image_stream.read()),
+            })
+            replacements[xmlid] = new.id
+    return replacements
+
+
+def noupdate_changes(env):
+    """Set some specific noupdate=1 records to noupdate=0"""
+    noupdates = env["ir.model.data"].search([
+        ("model", "=", "website"),
```

Comment:
`("model", "=", "ir.ui.view"),` 🤔

And not sure if also `("module", "=", "website"),` 🤔

Reply:
Yes, better for restricting (we can have that XML-ID in other modules).

Reply:
Thanks, it was a typo. Fixed now.